### PR TITLE
[176] [180] [76] Implement notification UI / Implement Casting notification UI /Fix change job's status UI

### PR DESCRIPTION
**Pull request by priasdevo**
## What did you do

- Fix responsive for JobMenu ( 76 sprint 2 task)
- image
- Implement Notification Page for Actor/Casting
- image
- Accept/Reject offer from notification Page
- ** Change NotificationsJobDto (status from string=>ApplicationStatus)

## Demo

- https://dev.modela.miello.dev/notification

### CODE REVIEW

File: apps/web/src/modules/job/components/JobMenu/styled.ts [hunk 1]
```diff
@@ -20,3 +22,13 @@ export const SummaryContainer = styled('div')`
...
+
+export const StatusButton = styled(Button)`
+  border-radius: '12px';
+  width: '100%';
+  font-size: '16px';
+  ${(props) => props.theme.breakpoints.down('lg')} {
+    width: '50%';
+    margin: 10px 30px 0 30px;
+  }
+`
```
Comment (miello) [inline apps/web/src/modules/job/components/JobMenu/styled.ts:34]:
```suggestion
export const StatusButton = styled(Button)`
  border-radius: 12px;
  width: 100%;
  font-size: 16px;
  ${(props) => props.theme.breakpoints.down('lg')} {
    width: 50%;
    margin: 10px 30px 0 30px;
  }
`
```

File: apps/web/src/modules/notification/components/NoticardContainer/components/NotiCard/Components/NotiCardBody/index.tsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+          <Link href={'/profile/' + actorId} underline="none">
+            {firstName + ' '}
+            {middleName + ' '}
+            {lastName}
+          </Link>
+        ) : userType === UserType.ACTOR &&
+          type === NotificationType.APPROVE_REFUND ? (
+          reason
+        ) : (
+          <Link href={'/job/' + jobId} underline="none">
```
Comment (miello) [inline apps/web/src/modules/notification/components/NoticardContainer/components/NotiCard/Components/NotiCardBody/index.tsx:39]:
```suggestion
          <Link href={`/job/${jobId}`} underline="none">
```

File: apps/web/src/modules/notification/components/NoticardContainer/components/NotiCard/Components/NotiCardBody/index.tsx [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+  return (
+    <BodyContainer>
+      <Typography variant="subtitle2" sx={{ color: 'rgba(0, 0, 0, 0.6)' }}>
+        {userType === UserType.ACTOR
+          ? actorBodyMap[type]
+          : castingBodyMap[type]}
+        {userType === UserType.CASTING &&
+        (type === NotificationType.ACCEPT_OFFER ||
+          type === NotificationType.REJECT_OFFER) ? (
+          <Link href={'/profile/' + actorId} underline="none">
```
Comment (miello) [inline apps/web/src/modules/notification/components/NoticardContainer/components/NotiCard/Components/NotiCardBody/index.tsx:30]:
```suggestion
          <Link href={`/profile/${actorId}`} underline="none">
```

File: apps/web/src/modules/notification/components/NoticardContainer/components/NotiCard/Components/NotiCardFooter/index.tsx [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+            }}
+            onClick={() => {
+              if (jobTitle) {
+                setTitle(jobTitle)
+              }
+              if (jobId) {
+                setFocusId(jobId)
+              }
+              openRejectModal()
+            }}
```
Comment (miello) [inline apps/web/src/modules/notification/components/NoticardContainer/components/NotiCard/Components/NotiCardFooter/index.tsx:44]: Consider making it another function

File: apps/web/src/modules/notification/components/NoticardContainer/components/NotiCard/Components/NotiCardFooter/index.tsx [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+            onClick={() => {
+              if (jobTitle) {
+                setTitle(jobTitle)
+              }
+              if (jobId) {
+                console.log('JobId : ', jobId)
+                setFocusId(jobId)
+              }
+              openAcceptModal()
```
Comment (miello) [inline apps/web/src/modules/notification/components/NoticardContainer/components/NotiCard/Components/NotiCardFooter/index.tsx:63]: Consider making it another function

File: apps/web/src/modules/notification/components/NoticardContainer/components/NotiCard/Components/NotiCardHeader/index.tsx [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+            : castingIconMap[type]}
+        </IconContainer>
+        <div>
+          <Typography variant="h6" sx={{ wordBreak: 'break-word' }}>
+            {userType === UserType.ACTOR
+              ? actorHeaderMap[type]
+              : castingHeaderMap[type]}
+            {userType === UserType.CASTING &&
+            (type === NotificationType.APPROVE_REFUND ||
+              type === NotificationType.REJECT_REFUND)
```
Comment (miello) [inline apps/web/src/modules/notification/components/NoticardContainer/components/NotiCard/Components/NotiCardHeader/index.tsx:41]: Repetitive condition checking to line 46, maybe making it as an another variable ?

File: apps/web/src/modules/notification/pages/index.tsx [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+    handleRejectModalSubmit,
+    handleAcceptModalOpen,
+    handleRejectModalOpen,
+    setFocusId,
+    title,
+    setTitle,
+  } = useModalData()
+  useNavbarSearch(
+    useCallback(() => {
+      console.log('TEST')
```
Comment (miello) [inline apps/web/src/modules/notification/pages/index.tsx:36]: ???

File: apps/web/src/modules/notification/components/RejectOfferModal/styled.ts [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+    background-color: #66a373af;
+  }
+`
+
+export const CancelButton = styled(Button)`
+  background-color: #aa5b5b;
+  &:hover {
+    background-color: #aa5b5baf;
+  }
+`
```
Comment (miello) [inline apps/web/src/modules/notification/components/RejectOfferModal/styled.ts:31]: Consider using `color="success"` or `color="error"` instead

File: apps/web/src/modules/notification/pages/hooks/useModalData/index.ts [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+    openRejectModal()
+  }, [openRejectModal])
+
+  const handleRejectCloseModal = useCallback(() => {
+    closeRejecttModal()
+  }, [closeRejecttModal])
+
+  const handleRejectModalSubmit = useCallback(async () => {
+    try {
+      await apiClient.put<string>(`/jobs/` + focusId + '/offer/reject')
```
Comment (miello) [inline apps/web/src/modules/notification/pages/hooks/useModalData/index.ts:48]:
```suggestion
      await apiClient.put<string>(`/jobs/${focusId}/offer/reject`)
```

**Review by miello — CHANGES_REQUESTED** [review]
This covers only code reviews

**Comment by miello** [comment]
Too much width ? (Account casting 10 and jobId 1)
image

**Comment by miello** [comment]
image

Maybe wrap to new line ?

**Comment by priasdevo** [comment]
@miello

**Review by miello — APPROVED** [review]

**Pull request merged**